test/extended/cli/builds.go

```diff
@@ -0,0 +1,339 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	t "github.com/onsi/gomega/types"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	bcOutputTemplate       = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	bcOutputToNameTemplate = "{{ .spec.output.to.name }}"
+	bcSourceTypeTemplate   = "{{ .spec.source.type }}"
+	testDockerFileContent  = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+func waitForBuild(oc *exutil.CLI, bcName, buildNumber string) error {
+	args := []string{"builds"}
+
+	if buildNumber != "" {
+		args = append(args, fmt.Sprintf("%v-%v", bcName, buildNumber))
+	} else {
+		args = append(args, "-l", fmt.Sprintf("openshift.io/build-config.name=%v", bcName))
+	}
+
+	return wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		out, err := oc.Run("get").Args(args...).Output()
+
+		if err != nil {
+			return false, nil
+		}
+		lines := strings.Split(out, "\n")
+
+		return len(lines) >= 2, nil
+	})
+}
+
+func getExpectedBCOutputMatcher(bcName string, isNames ...string) t.GomegaMatcher {
+	matcher := o.ContainSubstring("Success")
+	if bcName != "" {
+		matcher = o.And(matcher, o.ContainSubstring(fmt.Sprintf("buildconfig.build.openshift.io \"%v\" created", bcName)))
```

Comment:
`%q` will be more appropriate, it'll handle quoting the string for you.

Reply:
+1, looks much nicer